libevmcore/Assembly.cpp

```diff
@@ -156,6 +163,119 @@ ostream& Assembly::stream(ostream& _out, string const& _prefix, StringMap const&
 	return _out;
 }
 
+Json::Value Assembly::createJsonValue(string _name, int _begin, int _end, string _value, string _jumpType) const
+{
+	Json::Value value;
+	value["name"] = _name;
+	value["begin"] = _begin;
+	value["end"] = _end;
+	if (!_value.empty())
+		value["value"] = _value;
+	if (!_jumpType.empty())
+		value["jumpType"] = _jumpType;
+	return value;
+}
+
+string toStringInHex(u256 _value)
+{
+	std::stringstream hexStr;
+	hexStr << hex << _value;
+	return hexStr.str();
+}
+
+Json::Value Assembly::streamAsmJson(ostream& _out, string const& _prefix, StringMap const& _sourceCodes, bool _inJsonFormat) const
+{
+	Json::Value root;
+
+	string currentArrayName = ".code";
+	std::vector<Json::Value> currentCollection;
+	for (AssemblyItem const& i: m_items)
+	{
+		switch (i.type())
+		{
+		case Operation:
+			currentCollection.push_back(
+				createJsonValue(instructionInfo(i.instruction()).name, i.getLocation().start, i.getLocation().end, i.getJumpTypeAsString()));
+			break;
+		case Push:
+			currentCollection.push_back(
+				createJsonValue(string("PUSH"), i.getLocation().start, i.getLocation().end, toStringInHex(i.data()), i.getJumpTypeAsString()));
+			break;
+		case PushString:
+			currentCollection.push_back(
+				createJsonValue(string("PUSH tag"), i.getLocation().start, i.getLocation().end, m_strings.at((h256)i.data())));
+			break;
+		case PushTag:
+			currentCollection.push_back(
+				createJsonValue(string("PUSH [tag]"), i.getLocation().start, i.getLocation().end, toStringInHex(i.data())));
+			break;
+		case PushSub:
+			currentCollection.push_back(
+				createJsonValue(string("PUSH [$]"), i.getLocation().start, i.getLocation().end, dev::toString(h256(i.data()))));
+			break;
+		case PushSubSize:
+			currentCollection.push_back(
+				createJsonValue(string("PUSH #[$]"), i.getLocation().start, i.getLocation().end, dev::toString(h256(i.data()))));
+			break;
+		case PushProgramSize:
+			currentCollection.push_back(
+				createJsonValue(string("PUSHSIZE"), i.getLocation().start, i.getLocation().end));
+			break;
+		case Tag:
+		{
+			currentCollection.push_back(
+				createJsonValue(string("tag"), i.getLocation().start, i.getLocation().end, string(i.data())));
+			currentCollection.push_back(
+				createJsonValue(string("JUMDEST"), -1, -1));
```

Comment:
This should have the same start and end location as the tag (it actually _is_ the tag). You can also remove the braces now.